File: app/src/main/java/cc/smartcash/wallet/Services/WebWalletAPIService.java
package cc.smartcash.wallet.Services;

import java.util.List;

import cc.smartcash.wallet.Models.LoginResponse;
import cc.smartcash.wallet.Models.User;
import cc.smartcash.wallet.Models.UserRecoveryKey;
import cc.smartcash.wallet.Models.UserRegisterRequest;
import cc.smartcash.wallet.Models.WebWalletContact;
import cc.smartcash.wallet.Models.WebWalletRootResponse;
import cc.smartcash.wallet.Models.WebWalletUserAvailableRequest;
import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.GET;
import retrofit2.http.Header;
import retrofit2.http.POST;

public interface WebWalletAPIService {

    @POST("user/authenticate")
    @FormUrlEncoded
    Call<LoginResponse> getToken(@Field("username") String username,
                                 @Field("password") String password,
                                 @Field("grant_type") String grantType,
                                 @Field("client_id") String clientId,
                                 @Field("TwoFactorAuthentication") String twoFactorAuthentication,
                                 @Field("client_type") String clientType,
                                 @Field("client_ip") String clientIp,
                                 @Field("client_secret") String clientSecret);

    @GET("user/my")
    Call<WebWalletRootResponse<User>> getUser(@Header("Authorization") String auth);

    @GET("contact/my")
    Call<WebWalletRootResponse<List<WebWalletContact>>> getUserContacts(@Header("Authorization") String auth);

    @POST("user")
    Call<WebWalletRootResponse<User>> setUser(@Body UserRegisterRequest auth);

    @GET("user/newkey")
    Call<WebWalletRootResponse<UserRecoveryKey>> getNewMasterSecurityKey();

    @POST("user/available")
    Call<WebWalletRootResponse<Boolean>> isUserAvailable(@Body WebWalletUserAvailableRequest request);
}

File: app/src/main/java/cc/smartcash/smarthub/Services/WebWalletAPIService.kt
package cc.smartcash.smarthub.Services

import cc.smartcash.smarthub.Models.*
import retrofit2.Call
import retrofit2.http.*

interface WebWalletAPIService {

    @get:GET("user/newkey")
    val newMasterSecurityKey: Call<WebWalletRootResponse<UserRecoveryKey>>

    @POST("user/authenticate")
    @FormUrlEncoded
    fun getToken(@Field("username") username: String,
                 @Field("password") password: String,
                 @Field("grant_type") grantType: String,
                 @Field("client_id") clientId: String,
                 @Field("TwoFactorAuthentication") twoFactorAuthentication: String,
                 @Field("client_type") clientType: String,
                 @Field("client_ip") clientIp: String,
                 @Field("client_secret") clientSecret: String): Call<LoginResponse>

    @GET("user/my")
    fun getUser(@Header("Authorization") auth: String): Call<WebWalletRootResponse<User>>

    @GET("contact/my")
    fun getUserContacts(@Header("Authorization") auth: String): Call<WebWalletRootResponse<List<WebWalletContact>>>

    @POST("user")
    fun setUser(@Body auth: UserRegisterRequest): Call<WebWalletRootResponse<User>>

    @POST("user/available")
    fun isUserAvailable(@Body request: WebWalletUserAvailableRequest): Call<WebWalletRootResponse<Boolean>>
}
